test/Concurrency/actor_isolation.swift

```diff
@@ -118,7 +118,17 @@ protocol MainCounter {
 
 struct InferredFromConformance: MainCounter {
   var counter = 0
+  var ticker: Int { // this is ok, only because this is in a struct
+    get { 1 }
+    set {}
+  }
+}
+
+class InferredFromConformanceClass: MainCounter {
+  var counter = 0
+  // expected-warning@-1{{actor-isolated property 'counter' cannot be used to satisfy settable property protocol requirement}}
   var ticker: Int {
+    // expected-warning@-1{{actor-isolated property 'ticker' cannot be used to satisfy settable property protocol requirement}}
     get { 1 }
     set {}
   }
```

Comment:
I think we must preserve the behavior that this is not an error, because a global-actor property requirement shouldn't be required to be immutable. For example, if somebody tries to mutate the property from the wrong context, we already get an isolation diagnostic:

```
protocol HasMainThreadCounter {
  @MainActor var counter: Int { get set }
}

class Hello: HasMainThreadCounter {
  @MainActor var counter: Int {
    get { 10 }
    set { print("set!") }
  }
}

@MainActor func okay(_ h: any HasMainThreadCounter) -> any HasMainThreadCounter {
  var g = h
  g.counter = 20
  return g
}

func rejected_by_isolation_checking(_ h: any HasMainThreadCounter) -> any HasMainThreadCounter {
  var g = h
  g.counter = 20 // error: main actor-isolated property 'counter' can not be referenced from a non-isolated context
  return g
}
```

My feeling is that we might need to just extend that check to make sure it verifies an `any P` where `P : Actor` rejects all mutations through that existential, because we can't prove the correct isolation has been obtained. That is, if somebody wrote this:

```
protocol PA: Actor {
  var pa: { get set }
}

actor A: PA {
  var pa: Int = 10

  func increment() {
    let opaque: any PA = self
    opaque.pa += 1 // ERROR: typechecker can't prove `opaque : A` && `opaque == self`
    self.pa += 1 // OK, since we know the isolation is OK.
  }
}
```

Now, is there _value_ in allowing users to define a requirement `PA.pa` in an `Actor` protocol, when you basically can't access it through that protocol? I don't know. This requirement is basically the same as `var pa: Int { get private(set) }`, and private requirements don't make sense for a protocol.

Now, this all contrasts with global-actors, where we _can_ prove the context of the mutable access matches with the isolation requirement, because there's only one instance of a global-actor.

Reply:
I wonder if allowing this conformance like you show in your last example should be allowed or not... seems a bit weird, but perhaps makes sense?

We do have cases of APIs that you can declare but only make use of from certain contexts... Distributed actors have quite a lot of this too -- you can have them conform to not distributed things, but those things you can only call from the "inside" and not cross actor then... In a way, this is the exact same scenario: inside we know exactly it is fine to mutate; and outside we can only guarantee the get is fine...

If you'd have time to take this over I'd welcome that; I took a stab at it thinking it'll be simple but it seems there is more to be done here 👍